revapi-basic-features/src/main/java/org/revapi/basic/VersionsTransform.java

```diff
@@ -622,7 +621,19 @@ private static TextModification parseTextModification(JsonNode modificationNode)
                 return new TextModification(modificationNode.asText(), null, null);
             } else if (modificationNode.isObject()) {
                 String prepend = modificationNode.path("prepend").asText();
-                String append = modificationNode.path("append").asText();
+                String append = null;
+
+                if (modificationNode.path("append").isObject()) {
+                    if (modificationNode.path("append").hasNonNull("majorIncreaseRequired")) {
```

Comment:
@metlos sir I fixed it.
now we can modify the `append` message configured by the user, according to the version change.

also sir, can you tell me how to add a test case, I think I'm doing it wrong. because in test cases the condition is always for `PATCH` only. Do we need to add conditions for `MAJOR` &  `MINOR` too?